src/utils.js

```diff
@@ -691,16 +767,16 @@ function getCompileFn(loaderContext, implementation, options) {
 
             // Check again because awaiting the initialization function
             // introduces a race condition.
-            if (!sassModernCompilers.has(implementation)) {
-              sassModernCompilers.set(implementation, compiler);
-              webpackCompiler.hooks.watchClose.tap("sass-loader", () => {
+            if (!sassModernCompilers.has(webpackCompiler)) {
+              sassModernCompilers.set(webpackCompiler, compiler);
+              webpackCompiler.hooks.shutdown.tap("sass-loader", () => {
                 compiler.dispose();
               });
             }
           }
 
           return sassModernCompilers
-            .get(implementation)
+            .get(webpackCompiler)
```

Comment:
@alexander-akait the `sassModernCompilers` map should map to `implementation`, not `webpackCompiler`. The current `modern-compiler` API in 14.2.0 is broken and leads to my system becoming unresponsive; I think because of this change.

Reply:
> leads to my system becoming unresponsive

Please provide steps to reproduce

Reply:
> Please provide steps to reproduce

On a larger project, running webpack in watch mode with `modern-compiler`. I don't have detailed steps.

> because when you have two compilers with `sass-loader` (i.e. multi compiler mode) after closing first we close sass compiler, so the second will failed

If we do want to use `webpackCompiler` as the key, this line also needs updating:

https://github.com/webpack-contrib/sass-loader/blob/4edc139755541ee892da51412d61e40e1aa36a85/src/utils.js#L763

Reply:
I see, yeah, I will fix :+1: 

> On a larger project, running webpack in watch mode with modern-compiler. I don't have detailed steps.

Do you find a problem? I want to investigate your problem

Reply:
Fix will be soon

Reply:
@renspoesse Please try https://github.com/webpack-contrib/sass-loader/releases/tag/v14.2.1

Reply:
Sorry for delay

Reply:
I tried with the latest version and the issue still exists. I ran the debugger and stepped through sass-loader, and it looks like `webpackCompiler` doesn't maintain a stable reference across files; so `sassModernCompilers.has(implementation)` still isn't `true` and we keep creating a new Dart process for each file.

Reply:
> I tried with the latest version and the issue still exists. I ran the debugger and stepped through sass-loader, and it looks like webpackCompiler doesn't maintain a stable reference across invocations for different Sass files; so sassModernCompilers.has(implementation) still isn't true and we keep creating a new Dart process for each file.

this is not possible because we are creating the only one compiler perf configuration

Reply:
I need a reproducible test repo to undestand your problem, because compiler lives from beginning to end of your `watch`/`run`/`serve`

Reply:
Updating mini-css-extract-plugin from 1.6.0 to 2.9.0 resolved the issue.

Reply:
@renspoesse Great :star: